### PR TITLE
Adding UO Razor Script to available languages

### PR DESCRIPTION
**Pull request by jaseowns**
I play Ultima Online and we use a program called Razor to help script and hotkey certain actions.  I'm making a website related to these scripts and would love to be able to use the react-syntax-highlighter importing this language file from Prism.

This is my first time submitting anything like this to open source so let me know if I need to change anything!

Thanks,
Jaseowns

### CODE REVIEW

**Review by RunDevelopment — CHANGES_REQUESTED** [review]
Thank you for the PR @jaseowns!

I left you a few comments.

Regarding the order of tokens: Prism supports [greedy tokens](https://prismjs.com/extending.html#language-definitions). Strings and comments are typically greedy, so e.g. `" // foo "` is a string and `// " foo "` is a comment despite both containing the syntax of the other. Please make all comment and string greedy (`greedy: true`).
Also, please move all (the new) greedy tokens to the top of the grammar. This doesn't change the language, but it makes it faster.

File: components/prism-uorazor.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+Prism.languages.uorazor = {
+	'comment-hash': {
+		pattern: /(?:#.*$)/gm,
```
[the remark this answers is not in the document]
Comment (jaseowns) [reply components/prism-uorazor.js]: Thanks for the extra info, updated these and added greedy.

File: components/prism-uorazor.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		alias: 'comment'
+	},
+	'comment-slash': {
+		pattern: /(?:\/\/.*$)/gm,
```
[the remark this answers is not in the document]
Comment (jaseowns) [reply components/prism-uorazor.js]: Done

File: components/prism-uorazor.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		alias: 'comment'
+	},
+	'source-layers': {
+		pattern: /\b(?:arms|backpack|blue|bracelet|cancel|clear|cloak|criminal|earrings|enemy|facialhair|friend|friendly|gloves|gray|grey|ground|hair|head|innerlegs|innertorso|innocent|lefthand|middletorso|murderer|neck|nonfriendly|onehandedsecondary|outerlegs|outertorso|pants|red|righthand|ring|self|shirt|shoes|talisman|waist)\b/i,
+		alias: 'punctuation'
```
[the remark this answers is not in the document]
Comment (jaseowns) [reply components/prism-uorazor.js]: Thank you, I've updated these as functions.

File: components/prism-uorazor.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	'tag-name': {
+		pattern: /(^\{%-?\s*)\w+/,
+		lookbehind: true,
+		alias: 'keyword'
+	},
+	'delimiter': {
+		pattern: /^\{[{%]-?|-?[%}]\}$/,
+		alias: 'punctuation'
+	},
+	'string': {
```
Comment (RunDevelopment) [inline components/prism-uorazor.js:29]: `string` also has to be greedy. E.g. `say '#something'` is currently highlighted incorrectly.

Also, please move `string` up so it comes right after `comment-slash`. This won't change the language and improves performance.

Comment (jaseowns) [reply components/prism-uorazor.js]: Appreciate you reviewing and leaving clear instructions!  This has been moved/marked as greedy.

**Comment by RunDevelopment** [comment]
Thank you for contributing @jaseowns!

**Pull request merged**